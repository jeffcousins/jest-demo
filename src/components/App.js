import React from 'react';
import Header from './Header';

const App = ({ props }) => {
  return (
    <div className="container">
      <Header />
      <p className="new-p-tag">
        <img src="google.com/meme.jpg"></img>
        <span className="markup">
          Buttermilk pancakes
        </span>
      </p>
    </div>
  );
};

export default App;
