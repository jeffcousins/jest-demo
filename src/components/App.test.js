import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('should render App perfectly', () => {
  const component = renderer.create(<App />);
  const domTree = component.toJSON();

  expect(domTree).toMatchSnapshot();
});
