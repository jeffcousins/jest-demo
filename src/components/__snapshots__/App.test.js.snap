exports[`test should render App perfectly 1`] = `
<div
  className="container">
  <h2>
    OMG I FCKING LOVE JEST
  </h2>
  <p
    className="new-p-tag">
    <img
      src="google.com/meme.jpg" />
    <span
      className="markup">
      Buttermilk pancakes
    </span>
  </p>
</div>
`;
